apps/src/code-studio/initApp/loadApp.js

```diff
@@ -30,7 +40,181 @@ function showDisabledButtonsAlert(isHocScript) {
   ReactDOM.render(<DisabledBubblesAlert isHocScript={isHocScript}/>, div[0]);
 }
 
-module.exports = function (callback) {
+// Legacy Blockly initialization that was moved here from _blockly.html.haml.
+// Modifies `appOptions` with some default values in `baseOptions`.
+// TODO(dave): Move blockly-specific setup function out of shared and back into dashboard.
+function setupApp(appOptions) {
+  if (!window.dashboard) {
+    throw new Error('Assume existence of window.dashboard');
+  }
+  timing.startTiming('Puzzle', window.script_path, '');
+
+  var lastSavedProgram;
+
+  if (appOptions.hasContainedLevels) {
+    if (appOptions.readonlyWorkspace) {
+      // Lock the contained levels if this is a teacher viewing student work:
+      lockContainedLevelAnswers();
+    }
+    // Always mark the workspace as readonly when we have contained levels:
+    appOptions.readonlyWorkspace = true;
+  }
+
+  // Sets up default options and initializes blockly
+  var baseOptions = {
```

Comment:
Yeah, I think we know this to be painful/ugly/something we'd like to have a cleaner story around.